packages/manager/modules/pci/src/projects/project/storages/blocks/blocks.service.js

```diff
@@ -126,14 +128,23 @@ export default class PciProjectStorageBlockService {
           snapshots: this.getVolumeSnapshots(projectId, volume),
         }),
       )
-      .then(
-        ({ attachedTo, volume, snapshots }) =>
-          new BlockStorage({
-            ...volume,
-            attachedTo,
-            snapshots,
-          }),
-      );
+      .then(({ attachedTo, volume, snapshots }) => {
+        const localZones = this.getLocalZones(customerRegions);
+        return new BlockStorage({
+          ...volume,
+          attachedTo,
+          snapshots,
+          isLocalZone: localZones.some(
+            (region) => region.name === volume.region,
+          ),
```

Comment:
**nitpick:** Same logic and the constant is already maintained in lot of places. Please maintain it in one place and reuse across the PCI to have better maintainability.

Reply:
Hello @anooparveti, I have performed the refactor work.